src/timepicker/dd-timepicker.js

```diff
@@ -0,0 +1,213 @@
+
+angular.module('dd.ui.timepicker', [])
+    .directive('ddTimepicker', TimepickerDirective)
+    .service('timeparserService', timeparserService);
+
+var KEY_ENTER = 13, KEY_UP = 38, KEY_DOWN = 40;
+
+TimepickerDirective.$inject = ['timeparserService'];
+function TimepickerDirective(timeparserService) {
+
+    var directive = {
+        restrict: 'A',
+        require: 'ngModel',
+        replace: true,
+        scope: {
+            ngModel: '='
+        },
+        link: function (scope, element, attrs, ngModel) {
+            
+            //(view to model)
+            ngModel.$parsers.push(function (value) {
+                return timeparserService.toModel(value);
+            });
+            
+            //(model to view)
+            ngModel.$formatters.push(function (value) {
+                return timeparserService.toView(value);
+            });
+
+            element.bind('keydown keypress', function (event) {
+                if (event.which === KEY_ENTER && !ngModel.$modelValue) {
+                    updateViewValue(timeparserService.getFormattedTime());
+                    event.preventDefault();
+                } else if (event.which === KEY_UP) {
+                    updateViewValue(timeparserService.changeTime(ngModel.$modelValue, 1));
+                    event.preventDefault();
+                } else if (event.which === KEY_DOWN) {
+                    updateViewValue(timeparserService.changeTime(ngModel.$modelValue, -1));
+                    event.preventDefault();
+                }
+            });
+
+            element.bind('blur', function toModelTime() {
+                updateViewValue(timeparserService.toModel(ngModel.$modelValue));
+            });
+
+            function updateViewValue(value) {
+                ngModel.$setViewValue(value);
+                ngModel.$render();
+            }
+
+        }
+    };
+
+    return directive;
+
+}
+
+function timeparserService() {
+    var self = this;
+    
+    var amPmPattern = /^(\d+)(a|p)$/,
+        normalTimePattern = /^([0-9]|0[0-9]|1[0-9]|2[0-3])[.:][0-5][0-9]$/,
+        digitsPattern = /^[0-9]+$/;
+        
+    self.toModel = toModel;
```

Comment:
It's more clear to read which parts expose pulblic api. Example:

```javascript
self.getUser = getUser;
self.createUser = createUser;
selft.deleteUser = deleteUser;
```
Also this follows angular best practices.